rust_src/src/alloc.rs

```diff
@@ -57,7 +57,7 @@ pub fn bool_vector(args: &mut [LispObject]) -> LispObject {
     let vector = unsafe { make_uninit_bool_vector(args.len() as EmacsInt) };
 
     for (i, arg) in args.iter().enumerate() {
-        unsafe { bool_vector_set(vector, i as EmacsInt, arg.is_not_nil()) }
+        unsafe { bool_vector_set(vector, i as EmacsInt, !!*arg) }
```

Comment:
IMHO, `!!*arg` is so confusing and unreadable.

Reply:
```
diff --git a/rust_src/src/alloc.rs b/rust_src/src/alloc.rs
index 9ce09f8aad..ff099d9ffa 100644
--- a/rust_src/src/alloc.rs
+++ b/rust_src/src/alloc.rs
@@ -57,7 +57,7 @@ pub fn bool_vector(args: &mut [LispObject]) -> LispObject {
     let vector = unsafe { make_uninit_bool_vector(args.len() as EmacsInt) };

     for (i, arg) in args.iter().enumerate() {
-        unsafe { bool_vector_set(vector, i as EmacsInt, !!*arg) }
+        unsafe { bool_vector_set(vector, i as EmacsInt, bool!(*arg)) }
     }

     vector
diff --git a/rust_src/src/eval_macros.rs b/rust_src/src/eval_macros.rs
index 8e09736d9d..f24d507dab 100644
--- a/rust_src/src/eval_macros.rs
+++ b/rust_src/src/eval_macros.rs
@@ -7,6 +7,10 @@
  * the unsafe is not used.
  */

+macro_rules! bool {
+    ($symbol:expr) => { !!$symbol }
+}
+
 /// Macro to generate an error with a list from any number of arguments.
 /// Replaces xsignal0, etc. in the C layer.
 ///
```

That is one alternative. Then we have code like:

```
if bool!(some_lisp_val) {
   // do something
}
```